src/App.tsx

```diff
@@ -1,14 +1,60 @@
+/* eslint-disable @typescript-eslint/quotes */
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
-import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
-import { Loader } from './components/Loader';
+import { getTodos } from "./api";
+import { TodoList } from "./components/TodoList";
+import { TodoFilter } from "./components/TodoFilter";
+import { TodoModal } from "./components/TodoModal";
+import { Loader } from "./components/Loader";
+import { Todo } from "./types/Todo";
+import { Select } from "./types/Select";
+
+const filterTodos = (todos: Todo[], query: string, selectedCategory: Select) => {
+  let filteredTodos = [...todos];
+  const normalizedQuery = query.toLowerCase();
+
+  switch (selectedCategory) {
+    case Select.Completed:
+      filteredTodos = filteredTodos.filter((todo: Todo) => todo.completed);
+      break;
+    case Select.Active:
+      filteredTodos = filteredTodos.filter((todo: Todo) => !todo.completed);
+      break;
+    default:
+      break;
+  }
+
+  if (query) {
+    filteredTodos = filteredTodos
+      .filter((todo: Todo) => todo.title.toLowerCase()
+        .includes(normalizedQuery));
+  }
```

Comment:
Please, trim search value
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/54936044/40da2cb8-91ec-45ae-87e7-ac67638bd3ff)